packages/jsii/lib/assembler.ts

```diff
@@ -314,10 +321,10 @@ export class Assembler implements Emitter {
                 continue;
             }
             this._defer(() => {
-                if (!spec.isClassType(this._dereference(ref))) {
+                if (!spec.isClassType(this._dereference(ref, base.symbol.valueDeclaration))) {
                     this._diagnostic(base.symbol.valueDeclaration,
                                     ts.DiagnosticCategory.Error,
-                                    `Base type of ${jsiiType.fqn} is not a class (${spec.describeTypeReference(ref)})`);
+                                    `Base type of ${jsiiType.fqn} is not a class or cannot be dereferenced (${spec.describeTypeReference(ref)})`);
```

Comment:
Make this 'is not exported', that's more in line with what people are thinking about when writing code.

Reply:
Better yet, suppress the second error message by checking that result of _dereference is not `undefined`.

Reply:
✔️